### ISSUE TITLE
Incompatibility with code push: ngx-translate service not loaded + white screen on iPad 

### ISSUE THREAD

**Issue by polfernandez** [ionic-team/cordova-plugin-ionic-webview#31]
After uploading to the new WKWebview (https://ionicframework.com/docs/wkwebview/) I am experimenting an issue with the plugins ngx-translate and code-push.

When deploying the app with the cable connected to the iOS device (iPad with OX 9.3.5 / iPhone OX 11.0.3) everything goes fine (app works as expected). 
However when I upload it to the apple store and it gets accepted (release) I experiment 2 different behaviors:
- With the iPad iOS 9.3.5 the app does not load (white screen).
- With the iPhone iOS 11.0.3 the app loads but all the labels (from ngx-translate) don't appear (only their tags not the values).

Do I miss something in the translateLoader? I need to get the files from ./assets/i18n/. It is something related with the CORS?

```
// TranslateModule: AoT requires an exported function for factories
export function createTranslateLoader(http: Http) {
  return new TranslateHttpLoader(http, './assets/i18n/', '.json');
}
```

I already removed iOS platform and added it again. Removed ng-modules and reinstalled. 

My system info:

```
    @ionic/cli-utils  : 1.12.0
    ionic (Ionic CLI) : 3.12.0

global packages:

    cordova (Cordova CLI) : 6.5.0 

local packages:

    @ionic/app-scripts : 2.1.4
    Cordova Platforms  : ios 4.4.0
    Ionic Framework    : ionic-angular 3.6.1

System:

    ios-deploy : 1.9.2 
    ios-sim    : 6.0.0 
    Node       : v7.10.1
    npm        : 2.15.12 
    OS         : macOS Sierra
    Xcode      : Xcode 9.0 Build version 9A235 

Misc:

    backend : pro
```

**my plugins**

```
code-push 2.0.2-beta "CodePushAcquisition"
cordova-plugin-app-event 1.2.1 "Application Events"
cordova-plugin-camera 2.4.1 "Camera"
cordova-plugin-code-push 1.9.5-beta "CodePush"
cordova-plugin-compat 1.2.0 "Compat"
cordova-plugin-console 1.1.0 "Console"
cordova-plugin-device 1.1.6 "Device"
cordova-plugin-dialogs 1.3.3 "Notification"
cordova-plugin-file 4.3.3 "File"
cordova-plugin-file-transfer 1.6.3 "File Transfer"
cordova-plugin-firebase 0.1.24 "Google Firebase Plugin"
cordova-plugin-ionic-webview 1.1.16 "cordova-plugin-ionic-webview"
cordova-plugin-splashscreen 4.0.3 "Splashscreen"
cordova-plugin-statusbar 2.2.3 "StatusBar"
cordova-plugin-whitelist 1.3.2 "Whitelist"
cordova-plugin-zip 3.1.0 "cordova-plugin-zip"
cordova-sqlite-storage 2.0.4 "Cordova sqlite storage plugin"
de.appplant.cordova.plugin.local-notification 0.8.5 "LocalNotification"
ionic-plugin-keyboard 2.2.1 "Keyboard"
onesignal-cordova-plugin 2.2.2 "OneSignal Push Notifications"
```

[UPDATE]: when I remove cordova-plugin-code-push the ngx-translate and the app on iPad works fine.

my code push code:

```
 private _codePushSync() {
    if (process.env.CORDOVA_APP === true)) {
      this._codePush.sync().subscribe((syncStatus: SyncStatus) => {
        if (syncStatus === SyncStatus.INSTALLING_UPDATE) {
          this._translateService
            .get('general.codePush.updating')
            .subscribe((updating: string) => {
              this._ngZone.run(() => {
                const loading = this._loadingController.create({ content: updating });
                loading.present();
              });
            });
        }
      });
    }
  }
```

**Comment by robobobo** [comment]
There seems to be an issue with symbols when using this new ionic-webview plugin and code-push.

I can't actually complete 'ionic cordova build iOS' due to this error
`Undefined symbols for architecture arm64:
  "_OBJC_CLASS_$_CDVWKWebViewEngine", referenced from:
      l_OBJC_$_CATEGORY_CDVWKWebViewEngine_$_CodePush in CDVWKWebViewEngine+CodePush.o`

Code push works perfectly fine with the older [cordova-plugin-wkwebview-engine](https://github.com/apache/cordova-plugin-wkwebview-engine) plugin

**Comment by robobobo** [comment]
Issue logged with Code-push though I feel this is more an ionic issue [code-push issue #315](https://github.com/Microsoft/cordova-plugin-code-push/issues/315)

**Comment by ioncerize** [comment]
I had this problem for months, and then very recently I updated my dependecies and it went away; it turned out to be a cordova-plugin-code-push problem related to its dependency on the cordova-plugin-file plugin.  The Cordova plugins sub-dependencies management is awful in general, and it's very hard to coerce Cordova to install particular versions of subdependencies.  Seemingly related to that, I found that if I added cordova-plugin-code-push to my package.json and instead of formally adding the plugin I ran "cordova prepare" which detects and installs missing plugins, then the File sub-dependency wasn't being installed correctly and that was what was causing the whitescreen.  However, when I formally added the cordova-plugin-code-push via "cordova plugin add" then this subdependency was installed correctly and the whitescreen issue went away.
